### ISSUE TITLE
bug: stencil does not transpile tsconfig aliases on 3.4.1

### ISSUE THREAD

**Issue by mikaelkaron** [ionic-team/ionic-framework#27762]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

when upgrading to `7.1.2` I get a bunch of errors similar to this one during build

```
[ WARN  ]  Bundling Warning UNRESOLVED_IMPORT
           '@utils/helpers' is imported by ../../node_modules/@ionic/core/dist/collection/components/input/input.js,
           but could not be resolved – treating it as an external dependency
```

### Expected Behavior

no errors to display

### Steps to Reproduce

- install `@stencil/core@3.4.1`
- install `@ionic/core@7.1.2`
- import `@ionic/core` in global  `app.ts`
- `stencil build`

### Ionic Info

n/a (not using ionic CLI)

### Additional Information

this is a new regression in `7.1.2`, downgrading to `7.1.1` fixes the problem

**Comment by liamdebeasi** [comment]
Thanks for the report. It looks like there was a Stencil regression that caused `@utils/helpers` to not be transpiled to the correct relative path on build. I'll get in touch with the Stencil team to see if we can get this fixed on their end.

**Comment by liamdebeasi** [comment]
I am going to downgrade Stencil to 3.4.0 for now until the issue can be resolved on Stencil's side. Here is a dev build with the fix: `7.1.3-dev.11688736787.10e452c1`

**Comment by liamdebeasi** [comment]
The PR has been merged, and the next release of Ionic will have this fix. In the meantime, you can either downgrade to 7.1.1 or use the dev build I provided in https://github.com/ionic-team/ionic-framework/issues/27762#issuecomment-1625432351. Thanks for reporting this!
